### PR TITLE
Map vertical surfaces

### PR DESCRIPTION
**Pull request by haykam821**

### CODE REVIEW

**Review by YanisBft — APPROVED** [review]
Seems good, not sure about the package tho

**Comment by haykam821** [comment]
I placed it in the same package as `Direction`, and many classes are part of `util` rather than `world.gen` already.

**Comment by liach** [comment]
Hmm, is there a vertical surface class? We can create packages for them if needed, just like how game event types are put in their packages

**Comment by haykam821** [comment]
No, this is just an enum containing `FLOOR` and `CEILING`. I can name it `VerticalSurface` if that's better.

**Comment by liach** [comment]
Hmm, looking at `AxisDirection` `Axis` etc in `Direction` imo it's better to remove `Type` suffix.

**Review by liach — APPROVED** [review]
I doubt they may change it in the future. This name appears good.

Side note: Might need some manual merging with #2133

File: mappings/net/minecraft/class_5927.mapping [hunk 1]
```diff
@@ -0,0 +1,4 @@
+CLASS net/minecraft/class_5927
```
Comment (liach) [inline mappings/net/minecraft/class_5927.mapping]: Mind move this to `VegetationPatchFeatureConfig` instead?

File: mappings/net/minecraft/class_5931.mapping [hunk 2]
```diff
@@ -0,0 +1,4 @@
+CLASS net/minecraft/class_5931
```
Comment (liach) [inline mappings/net/minecraft/class_5931.mapping]: Same, now this is `CaveSurfaceDecoratorConfig`

File: mappings/net/minecraft/class_5927.mapping (hunk 1 above)
Comment (haykam821) [reply mappings/net/minecraft/class_5927.mapping]: Fixed in e762aacaed6c3345f0e94f4365d8d9d63f306ae8

File: mappings/net/minecraft/class_5931.mapping (hunk 2 above)
Comment (haykam821) [reply mappings/net/minecraft/class_5931.mapping]: Fixed in e762aacaed6c3345f0e94f4365d8d9d63f306ae8

**Pull request merged**